practice-two/src/components/common/Button.jsx

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+// Constanst
+import { BUTTON } from '../../constants/components';
+
+// Mapping the props to corresponding Tailwind CSS classes
+const variantClasses = {
+  primary:
+    `${BUTTON.DEFAULT} focus:border-secondary rounded-lg`,
+  secondary:
+    `${BUTTON.DEFAULT} focus:border-primary rounded-3xl`,
+};
+
+const sizeClasses = {
+  small: 'px-3 py-1 text-sm',
+  medium: 'px-4 py-2 text-base',
+  large: 'px-6 py-3 text-lg',
+};
+
+const colorClasses = {
+  green: 'text-green-500',
+  hello: 'text-blue-500',
+  red: 'text-red-500',
+  black: 'text-black-500',
+  gray: 'text-gray-500',
+};
+// eslint-disable-next-line react/function-component-definition
+
+/**
+ * Button component
+ * @param text text for button
+ * @param variant text for button
+ * @param size text for button
+ * @param rightIcon text for button
+ * @param image image
+ * @param rightIcon text for button
+ * @returns button element
+ */
+const Button = ({
+  text,
+  variant = 'primary',
+  size = 'medium',
+  rightIcon,
+  leftIcon,
+  image,
+  type = 'button',
+  color = 'black',
+  onClick = () => {},
+}) => {
+  const [iconState, setIconState] = useState('down');
+
+  const handleButtonClick = () => {
+    setIconState((prevState) => (prevState === 'down' ? 'up' : 'down'));
+    onClick();
+  };
+
+  const buttonClasses = [
+    'inline-flex items-center gap-2 pl-2 pr-2',
+    sizeClasses[size],
+    variantClasses[variant],
+    colorClasses[color],
+  ].join(' ');
+
+  const leftIconSrc = leftIcon ? 'public/images/Filter-icon.png' : null;
+  const rightIconSrc = rightIcon && `public/images/${iconState === 'down' ? 'up' : 'down'}.png`;
+
+  return (
+    <button type={type} className={buttonClasses} onClick={handleButtonClick}>
+      {leftIcon && <img src={leftIconSrc} alt="left_icon" className="w-5" />}

```

Comment:
Do you know the "Children" props. I think it's should be "Children". You can check tailwind button component, how they built

Reply:
Fixed it